### PR TITLE
Added nanobind to test suite

### PR DESCRIPTION
**Pull request by wjakob**
This PR adds nanobind to the PyPy test suite. Let's see 🤞

### CODE REVIEW

**Comment by wjakob** [comment]
Okay, looks like it's working! The tests are of course failing, but that's the failure I had reported.

**Comment by wjakob** [comment]
Curiously, there is a more complex failure pattern: some tests are reporting this error message
- https://github.com/pypy/binary-testing/actions/runs/6432935702/job/17468910164?pr=44
In other cases, the whole interpreter crashes without writing an error message
- https://github.com/pypy/binary-testing/actions/runs/6432935702/job/17468909828?pr=44

File: .github/workflows/nanobind.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python }}
+
+    - name: Update CMake
+      uses: jwlawson/actions-setup-cmake@v1.13
+
+    - name: Setup annotations on Linux
+      if: runner.os == 'Linux'
+      run: python -m pip install pytest-github-actions-annotate-failures
```
Comment (mattip) [inline .github/workflows/nanobind.yml:68]: This needs an `else` clause for the other platforms to install pytest

**Review by mattip — COMMENTED** [review]
It seems like pypy3.10 segfaults. Let's let the nightlies run and see if it still happens.

**Comment by wjakob** [comment]
Now everything fails consistently ;)

**Comment by mattip** [comment]
Reran the jobs to pick up new nightlies. There is now a new internal test failure from [a test with this comment](https://foss.heptapod.net/pypy/pypy/-/blob/branch/py3.9/pypy/module/cpyext/test/test_typeobject.py#L2218) “taken from https://github.com/wjakob/pypy_issues at commit 03890103”, maybe now PyPy behaves the same as cpython?

**Comment by mattip** [comment]
The windows build bot is down, so the results did not change. That 3.10 segfault is weird.

**Comment by mattip** [comment]
CI is green, merging. Thanks @wjakob

**Comment by wjakob** [comment]
Thank you 🙇

**Pull request merged**